lib/ddtrace/contrib/http/instrumentation.rb

```diff
@@ -37,22 +36,17 @@ module InstanceMethods
           def request(req, body = nil, &block)
             host, = host_and_port(req)
             request_options = datadog_configuration(host)
-            pin = datadog_pin(request_options)
-            return super(req, body, &block) unless pin
+            client_config = Datadog::Tracing.configuration_for(self)
 
             return super(req, body, &block) if Datadog::Contrib::HTTP.should_skip_tracing?(req)
 
             Datadog::Tracing.trace(Ext::SPAN_REQUEST, on_error: method(:annotate_span_with_error!)) do |span, trace|
               begin
-                # even though service_name might already be in request_options,
-                # we need to capture the name from the pin since it could be
-                # overridden
-                request_options[:service_name] = pin.service_name
-                span.service = service_name(host, request_options)
+                span.service = service_name(host, request_options, client_config)
                 span.span_type = Datadog::Ext::HTTP::TYPE_OUTBOUND
                 span.resource = req.method
 
-                if Datadog::Tracing.enabled? && !Datadog::Contrib::HTTP.should_skip_distributed_tracing?(pin)
+                if Datadog::Tracing.enabled? && !Datadog::Contrib::HTTP.should_skip_distributed_tracing?(client_config)
```

Comment:
This call to `Datadog::Contrib::HTTP.should_skip_distributed_tracing?` will raise an error if `client_config` is `nil`, which would happen if `client_config = Datadog::Tracing.configuration_for(self)` returns `nil` above.

Reply:
Nevermind, it looks like `should_skip_distributed_tracing?` has a `nil` check now.

Reply:
It shouldn't, as the method does `client_config && client_config.key?(:distributed_tracing)` internally before using it. I don't like how these methods were written originally, but I don't want to get too invested into cleaning them up right now.